tests/numpy_test.py

```diff
@@ -106,6 +109,146 @@ def test_ndarray_roundtrip(self):
+            npt.assert_array_equal(b, _b)
+
+    def test_transpose(self):
+        """test handling of non-c-contiguous memory layout"""
+        # simple case; view a c-contiguous array
+        a = np.arange(9).reshape(3, 3)
+        b = a[1:, 1:]
+        assert b.base is a.base
+        _a, _b = self.roundtrip([a, b])
+        assert _b.base is _a.base
+        npt.assert_array_equal(a, _a)
+        npt.assert_array_equal(b, _b)
+
+        # a and b both view the same contiguous array
+        a = np.arange(9).reshape(3, 3).T
+        b = a[1:, 1:]
+        assert b.base is a.base
+        _a, _b = self.roundtrip([a, b])
+        assert _b.base is _a.base
+        npt.assert_array_equal(a, _a)
+        npt.assert_array_equal(b, _b)
+
+        # view an f-contiguous array
+        a = a.copy()
+        a.strides = a.strides[::-1]
+        b = a[1:, 1:]
+        assert b.base is a
+        _a, _b = self.roundtrip([a, b])
+        assert _b.base is _a
+        npt.assert_array_equal(a, _a)
+        npt.assert_array_equal(b, _b)
+
+        # now a.data.contiguous is False; we have to make a deepcopy to make this work
+        a = np.arange(8).reshape(2, 2, 2).copy()
+        a.strides = a.strides[0], a.strides[2], a.strides[1]
+        b = a[1:, 1:]
+        assert b.base is a
+        with warnings.catch_warnings(record=True) as w:
+            _a, _b = self.roundtrip([a, b])
+            assert len(w) == 1
+            npt.assert_array_equal(a, _a)
+            npt.assert_array_equal(b, _b)
+
+    def test_buffer(self):
+        """test behavior with memoryviews which are not ndarrays"""
+        buffer = b'abcdefgh'
+        a = np.frombuffer(buffer, dtype=np.byte)
+        with warnings.catch_warnings(record=True) as w:
+            _a = self.roundtrip(a)
+            npt.assert_array_equal(a, _a)
+            assert len(w) == 1
+
+    def test_as_strided(self):
+        """test object with array interface which isnt an ndarray, like the result of as_strided"""
+        a = np.arange(10)
+        b = np.lib.stride_tricks.as_strided(a, shape=(5,), strides=(a.dtype.itemsize * 2,))
+        data = [a, b]
+
+        with warnings.catch_warnings(record=True) as w:
+            # as_strided returns a DummyArray object, which we can not currently serialize correctly
+            # FIXME: would be neat to add support for all objects implementing the __array_interface__
+            _data = self.roundtrip(data)
+            assert len(w) == 1
+
+        # as we were warned, deserialized result is no longer a view
+        with self.assertRaises(Exception):
+            _data[0][0] = -1
+            assert (_data[1][0] == -1)
+
+    def test_immutable(self):
+        """test that immutability flag is copied correctly"""
+
+    # def test_newstrided(self):
+    #     a = np.arange(10)
+    #     q = np.ndarray(buffer=a.data, strides=a.strides*2, shape=(5,6), dtype=a.dtype)
+    #     assert q.base is a
+    #     print(q)
+
```

Comment:
If we're going to have a commented-out test is it worth having a TODO comment explaining how we might handle it in the future?  Otherwise, a short comment explaining why it doesn't work(?) might help.  Ditto with the empty `test_immutable` above.